src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -175,6 +187,81 @@ func (c *controller) WaitAndUpdateNodesStatus(status *ControllerStatus) {
 	c.log.Infof("Done waiting for all the nodes. Nodes in error status: %d\n", hostsInError)
 }
 
+func (c *controller) hackDNSAddressConflict(wg *sync.WaitGroup) {
+
+	c.log.Infof("Making sure service %s can reserve the .10 address", dnsServiceName)
+
+	defer wg.Done()
+	networks, err := c.kc.GetServiceNetworks()
+	if err != nil || len(networks) == 0 {
+		c.log.Errorf("Failed to get service networks: %s", err)
+		return
+	}
+
+	ip, _, _ := net.ParseCIDR(networks[0])
+	ip4 := ip.To4()
+	if ip4 == nil {
+		c.log.Debugf("Service network is IPv6: %s", ip)
+		return
+	}
+	ip4[3] = 10 // .10 is the conflicting address
+
+	for i := 0; i < 45; i++ {
+		svs, err := c.kc.ListServices("")
+		if err != nil {
+			c.log.WithError(err).Warnf("Failed to list running services (#%d)", i+1)
+			time.Sleep(DNSAddressRetryInterval)
+			continue
+		}
+		s := c.findServiceByIP(ip4.String(), &svs.Items)
+		if s == nil {
+			c.log.Infof("No service found with IP %s (#%d)", ip4, i+1)
+			time.Sleep(DNSAddressRetryInterval)
+			continue
+		}
+		if s.Name == dnsServiceName && s.Namespace == dnsServiceNamespace {
+			c.log.Infof("Service %s has successfully taken IP %s", dnsServiceName, ip4)
+			break
+		}
+		c.log.Warnf("Deleting service %s in namespace %s whose IP %s conflicts with %s", s.Name, s.Namespace, ip4, dnsServiceName)
+		if err := c.killConflictingService(s); err != nil {
+			c.log.WithError(err).Warnf("Failed to delete service %s in namespace %s", s.Name, s.Namespace)
+		}
+		if err := c.deleteDNSOperatorPods(); err != nil {
```

Comment:
it must be deleted? it doesn't try to recreate service?

Reply:
The DNS service? Seems it doesn't. The original Ansible workaround is very specific about deleting DNS operator pods so that the service can be recreated with the right IP address.

Reply:
can we test it? cause from logs it looks like it tries

Reply:
another question is regarding service we kill. In case operator is not recreating its resources we need to restart operator that its service we had killed, right?

Reply:
I think the assumption is that K8s will re-create the service with another IP.

Reply:
As I said, I've been following the original workaround procedure. IMO it would take too much effort to come up with something better.